examples/crd_apply.rs

```diff
@@ -31,21 +31,25 @@ pub struct FooStatus {
     is_bad: bool,
     replicas: i32,
 }
-
+fn make_patch<T>(obj: T) -> Patch<T> {
+    Patch::Apply {
+        patch: obj,
+        field_manager: "crd_apply_example".to_string(),
+        force: true,
+    }
+}
```

Comment:
maybe a builder for `Patch` now makes sense if we have to create this :thinking:

Reply:
Hm, thinking about this more. In repeat `.patch` scenarios like this example, it's now no longer possible to store the parameters early on in a fn, because the data struct now bundles params, and this is unexpected compared to the rest of the Api. I'm starting to think that coupling the parameters with the enum - while introducing a small level of safety - isn't a good idea.

We could make the enum less awkward here by having builders, but users still have to go through something like `Patch::apply(patch, field_manager)` or `Patch::apply(patch).manager(mgr).build()` every time, when it's quite common to separate the data and the parameters.

I think the following style would introduce the least visual clutter, while retaining the most safety:

```rust
let serverside = PatchParams::apply(field_mgr);
pods.patch("foos.clux.dev", &serverside, Patch::apply(Foo::crd()))
```